pennylane/qnn/__init__.py

```diff
@@ -11,9 +11,19 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""This module contains classes and functions for constructing quantum neural networks from QNodes."""
+"""This module contains classes and functions for constructing quantum neural networks from
+QNodes."""
+import importlib
 
 import pennylane.qnn.cost
 
-from .keras import KerasLayer
-from .torch import TorchLayer
+class_map = {"KerasLayer": "keras", "TorchLayer": "torch"}
+
+
+def __getattr__(name):
+    """Allow for lazy-loading of KerasLayer and TorchLayer so that TensorFlow and PyTorch are not
+    automatically loaded with PennyLane"""
+    if name in class_map:
+        mod = importlib.import_module("." + class_map[name], __name__)
+        return getattr(mod, name)
+    raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
```

Comment:
Out of curiosity, what happens if we add

```python
__all__ = ["KerasLayer", "TorchLayer"]
```

here? Does it fix autocorrect? Does it break the lazy loading?

Reply:
What does the `!r` do?

Reply:
It calls the `repr()` of the object. In this case it gives the single inverted commas:

```pycon
>>> print(f"module is {__name__!r}")
module is '__main__'
```

```pycon
>>> print(f"module is {__name__}")
module is __main__
```

These are needed to fit the standard style:
```
AttributeError: module 'pennylane.qnn' has no attribute 'random'
```